psh/test-ls-rootfs.py

```diff
@@ -0,0 +1,76 @@
+# Phoenix-RTOS
+#
+# phoenix-rtos-tests
+#
+# additional psh ls command test for targets with root file system
+#
+# Copyright 2022 Phoenix Systems
+# Author: Damian Loewnau
+#
+# This file is part of Phoenix-RTOS.
+#
+# %LICENSE%
+#
+
+import psh.tools.psh as psh
+
+
+def assert_ls_pshcmds(p):
+    psh_cmds = psh.get_commands(p)
+
+    psh_cmd_pattern = r'(\x1b\[\S*)('
+    for psh_cmd in psh_cmds:
+        if psh_cmd != psh_cmds[0]:
+            psh_cmd_pattern += r'|'
+        psh_cmd_pattern += psh_cmd
+    psh_cmd_pattern += r')((\s)|(\x1b\[\S*))'
+
+    cmd = 'ls bin'
+    listed_psh_cmds = []
+    p.sendline(cmd)
+    p.expect(cmd)
+    idx = 0
+    while True:
+        idx = p.expect([psh.PROMPT, psh_cmd_pattern])
+        if idx == 1:
+            groups = p.match.groups()
+            if groups[1] not in listed_psh_cmds:
+                listed_psh_cmds.append(groups[1])
+        elif idx == 0:
+            break
+    msg = (f"The number of psh commands listed in bin is incorrect: {len(listed_psh_cmds)} instead of {len(psh_cmds)}."
+           + f"\nListed psh commands: {sorted(listed_psh_cmds)}"
+           + f"\nPsh commands got from help: {sorted(psh_cmds)}")
+
+    assert len(listed_psh_cmds) == len(psh_cmds), msg
```

Comment:
I would like to leave this case, because there may occur some problem (like in https://github.com/phoenix-rtos/phoenix-rtos-project/issues/194). But I didn't have idea how to do it better than avoiding `exit` and `history` for now.